iceoryx_examples/icedelivery/iox_subscriber_with_history.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020 by Apex.AI. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/popo/typed_subscriber.hpp"
+#include "iceoryx_posh/popo/user_trigger.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "topic_data.hpp"
+
+#include <csignal>
+#include <iostream>
+
+bool killswitch = false;
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT, now exit gracefully
+    killswitch = true;
+}
+
+int main()
+{
+    // register sigHandler for SIGINT
+    signal(SIGINT, sigHandler);
+
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime("iox-ex-subscriber-typed");
+
+    // initialized subscriber
+    iox::popo::SubscriberOptions subscriberOptions;
+    subscriberOptions.queueCapacity = 10U;
+    // The publisher is sending faster then we receive since we are waiting
+    // for one second in every loop cycle. Therefore we require a little bit of
+    // space for our samples in the history.
```

Comment:
This is a bit confusing. For the history it doesn't matter if the publisher frequency is higher. It is relevant how many samples the publisher has sent before the subscription became active. From these samples the subscriber can get some on subscribe. There is no more functionality for the history. If the publisher has a higher frequency, the queue size is your friend ;-)
If you really see the history depends on when you start the executables. Therefore I mention that it could make sense to demonstrate it in one process where we can control the delay between starting the publisher and the subscriber thread

Reply:
fixed